libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -604,7 +604,7 @@ def semantic_hybrid_search_with_score_and_rerank(
                             if result.get("@search.captions")
                             else {},
                             "answers": semantic_answers_dict.get(
-                                json.loads(result["metadata"]).get("key"),
+                                result.get(FIELDS_ID, ""),
```

Comment:
Should this be `FIELDS_TAG`?

Reply:
Yes, you're right - my response was a knee-jerk reaction to my particular related issue, rather than the OPs exact issue.
It's a bit confusing as the term "metadata" gets thrown around a lot - sometimes in a generic context, and sometimes in a very specific way - sometimes referring to anything that's not the main content (like an id), and sometimes referring to only user-supplied, extra meta-data

Reply:
No worries, I understand the confusion 😃 I'll mark this thread as resolved.